test/functional/config/utils.go

```diff
@@ -163,97 +174,100 @@ func (u *TestUtils) runCmd(cmdline string) (string, error) {
 }
 
-func (u *TestUtils) AwaitCertReady(names ...string) error {
-	return u.AwaitState("cert", "Ready", names...)
+func (u *TestUtils) WaitUntilCertReady(ctx context.Context, names ...string) error {
+	return u.WaitUntilCertState(ctx, "Ready", names...)
 }
 
-func (u *TestUtils) AwaitCertError(names ...string) error {
-	return u.AwaitState("cert", "Error", names...)
+func (u *TestUtils) WaitUntilCertError(ctx context.Context, names ...string) error {
+	return u.WaitUntilCertState(ctx, "Error", names...)
 }
 
-func (u *TestUtils) AwaitCertDeleted(names ...string) error {
-	return u.AwaitState("cert", STATE_DELETED, names...)
+func (u *TestUtils) WaitUntilCertDeleted(ctx context.Context, names ...string) error {
+	return u.WaitUntilCertState(ctx, STATE_DELETED, names...)
 }
 
-func (u *TestUtils) AwaitCertRevoked(names ...string) error {
-	return u.AwaitState("cert", "Revoked", names...)
+func (u *TestUtils) WaitUntilCertRevoked(ctx context.Context, names ...string) error {
+	return u.WaitUntilCertState(ctx, "Revoked", names...)
 }
 
-func (u *TestUtils) AwaitCertRevocationApplied(name string) error {
-	return u.AwaitState("certrevoke", "Applied", name)
+func (u *TestUtils) WaitUntilCertRevocationApplied(ctx context.Context, name string) error {
+	return u.WaitUntilCertificateRevocationState(ctx, "Applied", name)
 }
 
-func (u *TestUtils) AwaitState(resourceName, expectedState string, names ...string) error {
-	msg := fmt.Sprintf("%s not %s: %v", resourceName, expectedState, names)
-	return u.Await(msg, func() (bool, error) {
-		output, err := u.runKubeCtl("get " + resourceName + " \"-o=jsonpath={range .items[*]}{.metadata.name}={.status.state}{'\\n'}{end}\"")
-		if err != nil {
-			return false, err
+// WaitUntilCertState takes names of Certificates and waits for them to get ready with a timeout of 15 seconds.
+func (u *TestUtils) WaitUntilCertState(ctx context.Context, expectedState string, names ...string) error {
+	return waitUntil(ctx, u.Client, &v1alpha1.Certificate{}, func(obj *v1alpha1.Certificate) (bool, error) {
+		if obj.Status.State != expectedState {
+			return retry.MinorError(fmt.Errorf("Certificate %s has state %s != %s", obj.Name, obj.Status.State, expectedState))
 		}
+		return retry.Ok()
 
-		states := map[string]string{}
-		lines := strings.Split(output, "\n")
-		for _, line := range lines {
-			cols := strings.Split(line, "=")
-			if len(cols) == 2 {
-				states[cols[0]] = cols[1]
-			}
-		}
-		for _, name := range names {
-			if expectedState == STATE_DELETED {
-				if _, ok := states[name]; ok {
-					return false, nil
-				}
-			} else if states[name] != expectedState {
-				return false, nil
-			}
-		}
-		return true, nil
-	})
+	}, u.AwaitTimeout, expectedState == STATE_DELETED, u.Namespace, names...)
 }
 
-type CheckFunc func() (bool, error)
+// WaitUntilCertificateRevocationState takes names of CertificateRevocations and waits for them to get ready with a timeout of 15 seconds.
```

Comment:
Same as above if accepted
```suggestion
// WaitUntilCertificateRevocationState takes the names of CertificateRevocations and waits for them to get ready with a timeout.
```